qiskit_experiments/data_processing/nodes.py

```diff
@@ -555,3 +560,154 @@ def _process(self, data: np.ndarray) -> np.ndarray:
             The data that has been processed.
         """
         return 2 * (0.5 - data)
+
+
+class RestlessNode(DataAction, ABC):
+    """An abstract node for restless data processing nodes."""
```

Comment:
This needs a bit more doc.
```suggestion
    """An abstract node for restless data processing nodes.
    
    In restless measurements, the qubit is not reset after each measurement. Instead, the
    outcome of the previous quantum non-demolition measurement is the initial state for the
    current circuit. Restless measurements therefore require special data processing nodes
    that are implemented as sub-classes of `ResltessNode`. 
    """
```